File: JS/problem004.js

// A palindromic number reads the same both ways. The largest palindrome made
// from the product of two 2-digit numbers is 9009 = 91 × 99.
//
// Find the largest palindrome made from the product of two 3-digit numbers.

for(a=b=1e2,max=0;(c=a*b)&&a<1e3;b<1e3?b++:a++&&(b=a))if((c+'')==(c+'').split('').reverse().join('')&&c>max)max=c;max;

File: c++/problem004.cpp
#include <iostream>
#include <string>

// problem004.cpp
// Find the largest palindrome made from the product of two 3-digit numbers

bool isPalindrome(std::string str) {
	for (int i = 0, len = str.length(); i < len; i++) {
		if (str[i] != str[len - 1 - i]) { return false; }
	}
	return true;
}

int main() {
	int max = 0;

	for (int i = 100, j = 100, num; i < 1000; j < 1000 ? i : i++, j < 1000 ? j++ : j = 100) {
		num = i * j;
		if (num > max && isPalindrome(std::to_string(num))) {max = num;}
	}

	std::cout << max << std::endl;
	return 0;
}

